app/server/lib/AttachmentStoreProvider.ts

```diff
@@ -0,0 +1,64 @@
+import { IAttachmentStore } from "app/server/lib/AttachmentStore";
+import log from 'app/server/lib/log';
+
+export type AttachmentStoreId = string
+
+export interface IAttachmentStoreProvider {
+  // Returns the store associated with the given id, returning null if no store with that id exists.
+  getStore(id: AttachmentStoreId): Promise<IAttachmentStore | null>
+
+  getAllStores(): Promise<IAttachmentStore[]>;
+
+  storeExists(id: AttachmentStoreId): Promise<boolean>;
+}
+
+export interface IAttachmentStoreBackendFactory {
+  name: string,
+  create: (storeId: string) => IAttachmentStore,
+}
+
+interface IAttachmentStoreDetails {
+  id: string;
+  factory: IAttachmentStoreBackendFactory;
+}
+
+export class AttachmentStoreProvider implements IAttachmentStoreProvider {
```

Comment:
Have I seen a description of how stores work somewhere in this PR? For devs to have a mental model? Here might be a good place if not?

Reply:
I don't think I put an overview in the code, I'll add one here :+1: